include/boost/pfr/core.hpp

```diff
@@ -77,19 +77,24 @@ constexpr auto get(T&& val, std::enable_if_t< std::is_rvalue_reference<T&&>::val
 /// \code
 ///     std::vector< boost::pfr::tuple_element<0, my_structure>::type > v;
 /// \endcode
+/// \note The behavior of a program that adds specializations for tuple_element is undefined.
 template <std::size_t I, class T>
-using tuple_element = detail::sequence_tuple::tuple_element<I, decltype( ::boost::pfr::detail::tie_as_tuple(std::declval<T&>()) ) >;
+struct tuple_element : detail::sequence_tuple::tuple_element<I, decltype( ::boost::pfr::detail::tie_as_tuple(std::declval<T&>()) ) > {};
 
+template<std::size_t I, class T>
+struct tuple_element<I, const T> {
+    using type = std::add_const_t<typename boost::pfr::tuple_element<I, T>::type>;
```

Comment:
To avoid additional template instantiations:
```suggestion
    using type = const typename boost::pfr::tuple_element<I, T>::type;
```